conans/client/tools/scm.py

```diff
@@ -63,6 +63,14 @@ def clone(self, url, branch=None):
             branch_cmd = "--branch %s" % branch if branch else ""
             output = self.run('clone "%s" . %s' % (url, branch_cmd))
             output += self._configure_ssl_verify()
+
+        if submodule is "shallow":
```

Comment:
You should use ``submodule == "shallow"``. Your current imlementation relies on short string optimizations, but is platform and implementation dependent.

Reply:
Good to know! Thanks.